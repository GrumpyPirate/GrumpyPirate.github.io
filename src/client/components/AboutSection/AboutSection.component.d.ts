import { ChildrenProps } from 'types';

interface AboutSectionProps extends ChildrenProps {
  title: string;
}

export { AboutSectionProps };
