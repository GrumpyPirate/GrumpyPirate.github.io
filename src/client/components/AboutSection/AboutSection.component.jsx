import React from 'react';
import PropTypes from 'prop-types';

import Heading from 'components/Typography/Heading/Heading.component';

import classes from './AboutSection.component.scss';

const AboutSection = ({ title, children }) => (
  <div className={classes['about__section']}>
    <Heading level={2} displayLevel={3} text={title} className={classes['about__section__title']} />
    {children}
  </div>
);

AboutSection.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default AboutSection;
