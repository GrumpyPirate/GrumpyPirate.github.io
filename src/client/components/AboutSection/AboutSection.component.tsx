import React, { FunctionComponent } from 'react';

import Heading from 'components/Typography/Heading/Heading.component';

import { AboutSectionProps } from './AboutSection.component.d';
import classes from './AboutSection.component.scss';

const AboutSection: FunctionComponent<AboutSectionProps> = ({
  title,
  children,
}: AboutSectionProps) => (
  <div className={classes['about__section']}>
    <Heading level={2} displayLevel={3} text={title} className={classes['about__section__title']} />
    {children}
  </div>
);

export default AboutSection;
