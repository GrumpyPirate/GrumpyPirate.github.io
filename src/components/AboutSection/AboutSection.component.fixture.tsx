import React from 'react';

export default {
  title: 'Consequuntur consequatur inventore nam, laborum',
  children: <div>Test</div>,
};
