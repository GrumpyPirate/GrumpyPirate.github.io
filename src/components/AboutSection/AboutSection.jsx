// React
import React, { Component } from 'react'

// Components
// import Icon from '../Icons/Icons'

// SCSS
import './AboutSection.scss'

// Define AboutSection
class AboutSection extends Component {
  constructor(props) {
    super(props)
  } // /constructor(props)

  render () {
    return (
      <div className="about__section">
        <h2 className="h3 about__section__title mb-3">{this.props.title}</h2>

        {this.props.children}
      </div>
    )
  } // /render ()
} // /class AboutSection extends React.Component

export default AboutSection
