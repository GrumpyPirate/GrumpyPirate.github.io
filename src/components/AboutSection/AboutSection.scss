/* -------------------------------------------------------
** About Section
** -----------------------------------------------------*/
@import 'functions';
@import 'variables';
@import 'mixins';

.about {
  &__section {
    $title-icon-size: rem(32px);
    // $section-border-color: darken($body-bg, 25%);
    $section-border-color: mix($body-bg, palette(theme-accent-1), 75%);
    background-color: darken($body-bg, 2.5%);
    padding: $grid-gutter-width / 2;
    // box-shadow: 0 rem(2px) rem(2px) 0 rgba(0, 0, 0, .175);

    &__title {
      border-bottom: solid $hr-border-width $section-border-color;
      padding-bottom: $headings-margin-bottom;

      // Variant - has-icon
      &--has-icon {
        position: relative;
        padding-right: $title-icon-size;

        .about__section__title-icon {
          display: inline-block;
          position: absolute;
          top: 0;
          right: 0;
        } // /.about__section__icon
      } // /&--has-icon
    } // /&__title

    &__title-icon {
      display: inline-block;
      text-align: center;
      color: $headings-color;

      .icon {
        width: 1.125em;
        height: 1.125em;
      } // /.icon
    } // /&__title-icon, &__section-icon

    &__graphic {
      display: flex;
      flex-flow: row wrap;
      align-items: center;
      justify-content: center;

      &__icon {
        display: block;
        font-size: $font-size-h3;
        line-height: 1;
        text-align: center;
        flex: 0 1 auto;
        padding: 0 ($grid-gutter-width / 2);
        margin-bottom: $paragraph-margin-bottom;

        // md up
        @include media-breakpoint-up(md) {
          font-size: $font-size-h1 * 1.5;
        } // /md up

        // Variant - large
        &--large {
          @include media-breakpoint-up(md) {
            font-size: $font-size-h1 * 2;
          } // /md up
        } // /&--large
      } // /&__icon
    } // /&__graphic

    &:not(:last-of-type) {
      margin-bottom: $paragraph-margin-bottom;
      padding-bottom: $paragraph-margin-bottom;
      // border-bottom: solid $hr-border-width $section-border-color;

      // md up
      @include media-breakpoint-up(md) {
        margin-bottom: 3rem;
      } // /md up
    } // /&:not(:last-of-type)
  } // /&__section
} // /.about
