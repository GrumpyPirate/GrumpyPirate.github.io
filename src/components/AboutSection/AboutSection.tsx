import React, { FunctionComponent } from 'react';

import Heading from 'components/Typography/Heading/Heading';

import classes from './AboutSection.scss';
import { AboutSectionProps } from './AboutSection.types';

const AboutSection: FunctionComponent<AboutSectionProps> = ({ title, children }) => (
  <div className={classes['about__section']}>
    <Heading level={2} displayLevel={3} text={title} className={classes['about__section__title']} />
    {children}
  </div>
);

export default AboutSection;
