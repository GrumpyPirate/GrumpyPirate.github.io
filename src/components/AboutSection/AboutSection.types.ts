import { ChildrenProps } from 'types/common';

export interface AboutSectionProps extends ChildrenProps {
  title: string;
}
