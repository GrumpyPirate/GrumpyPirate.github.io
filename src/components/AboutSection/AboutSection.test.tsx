import React from 'react';
import { shallow } from 'enzyme';

import AboutSection from './AboutSection';

import fixture from './AboutSection.fixture';

describe('Components', () => {
  describe('AboutSection', () => {
    const props = {
      ...fixture,
    };

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<AboutSection {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
