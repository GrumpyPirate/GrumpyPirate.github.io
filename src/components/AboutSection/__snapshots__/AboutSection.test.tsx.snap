// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components AboutSection should render as expected, without crashing 1`] = `
<div
  className="about__section"
>
  <Heading
    className="about__section__title"
    displayLevel={3}
    level={2}
    text="Consequuntur consequatur inventore nam, laborum"
  />
  <div>
    Test
  </div>
</div>
`;
